### PR TITLE
Improving FD docstrings and adding comments

### PR DESCRIPTION
**Pull request by DanHickstein**
Just a few quick changes to clarify the Docstrings in FD.py, and to make the docstrings closer to the 80 character recommendation in PEP. 

Also, I added a few comments and formatting tweaks to make FD easier to understand. I'm not too attached to any of these changes, so feel free to merge and remix to your own tastes.

### CODE REVIEW

File: EMpy/modesolvers/FD.py [hunk 1]
```diff
@@ -329,17 +336,17 @@ def __init__(self, wl, x, y, epsfunc, boundary):
     def _get_eps(self, xc, yc):
         tmp = self.epsfunc(xc, yc)
 
-        def _reshape(tmp):
+        def _reshape(tmp): # pad the array by duplicating edge values
```
Comment (lbolla) [inline EMpy/modesolvers/FD.py]: This is not PEP8 compliant. I know that at the moment EMpy is not very PEP8 compliant at all, but new code should be.
Can you turn this comment into a docstring instead?

File: EMpy/modesolvers/FD.py [hunk 2]
```diff
@@ -545,16 +572,16 @@ def build_matrix(self):
         else:
             raise ValueError('unknown boundary conditions')
 
-        axxn[ib] += sign * axxs[ib]
+        axxn[ib]  += sign * axxs[ib]
```
Comment (lbolla) [inline EMpy/modesolvers/FD.py]: These are not PEP8 compliant either, but I see the aesthetic value.

File: EMpy/modesolvers/FD.py [hunk 3]
```diff
@@ -981,13 +986,14 @@ def solve(self, neigs=4, tol=0, guess=None):
         else:
             shift = None
 
+        # Here is where the actual mode-solving takes place! 
         [eigvals, eigvecs] = eigen.eigs(A,
-                                        k=neigs,
```
Comment (lbolla) [inline EMpy/modesolvers/FD.py]: Again, this is not PEP8 compliant. In this case, I think it's more elegant to leave no spaces around the "equal" signs.

**Comment by DanHickstein** [comment]
Thanks for the suggestions! I'm not really smart enough to use git on the command line. I just use the GitHub Desktop GUI thing. I tried to get the rebase to work, but I couldn't see the merge-commits on the command line.

I incorporated the rest of the suggestion.

**Comment by lbolla** [comment]
Thanks!

**Comment by DanHickstein** [comment]
Thanks for your help and your efficient merge :)

**Pull request merged**